core/src/main/java/io/aiven/kafka/tieredstorage/manifest/SegmentManifestV1.java

```diff
@@ -41,19 +41,19 @@ public SegmentManifestV1(
         @JsonProperty(value = "compression", required = true) final boolean compression,
         @JsonProperty("encryption") final SegmentEncryptionMetadataV1 encryption
     ) {
-        this(chunkIndex, segmentIndexes, compression, encryption, null);
+        this(chunkIndex, segmentIndexes, compression, Optional.ofNullable(encryption), null);
     }
 
     public SegmentManifestV1(final ChunkIndex chunkIndex,
                              final SegmentIndexesV1 segmentIndexes,
                              final boolean compression,
-                             final SegmentEncryptionMetadataV1 encryption,
+                             final Optional<SegmentEncryptionMetadataV1> encryption,
```

Comment:
Not that I'm a dogmatist, but even Brian Goetz [advises](https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555) against using `Optional` as args :smiley:

Reply:
That's fair. I was on the fence about this one. For private methods (where I started introducing it) looks like a better input type--even though as Brian mentions it's not intended as a "Maybe" type (not serializable, etc)--but it leaked to this public method. 
I'm ok falling back to the previous type. Do you think is still valid type for private methods or should we undo those too?

Reply:
I have undo changes for internal methods as well. In the case of SegmentManifestV1, I think it's a good candidate to introduce a Builder for it.
I have also spot a couple of input parameters using Optional in our code base, will do a quick refactor later to clean those up.

Reply:
Follow ups:
- https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/pull/532 to clean existing optional input types
- https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/pull/533 to have a manifest builder